app/web/src/features/dashboard/CommunityBrowser.tsx

```diff
@@ -0,0 +1,186 @@
+    selected?.[selected.length - 1]?.communityId || 1
+  );
+  const [cachedQueryResults, setCachedQueryResults] = useState<
+    {
+      data: Community.AsObject[];
+      hasMore?: boolean;
+    }[]
+  >([]);
+
+  const lastColumnRef = useRef<HTMLDivElement>(null);
+  useEffect(() => {
+    setTimeout(
+      () =>
+        lastColumnRef.current?.scrollIntoView({
+          behavior: "smooth",
+          block: "end",
+        }),
+      50
+    );
+  }, [selected]);
+
+  const handleClick = (community: Community.AsObject, level: number) => {
+    //if the last column is clicked
+    if (level === selected.length) {
+      setCachedQueryResults([
+        ...cachedQueryResults,
+        {
+          data: query.data!.pages.flatMap((page) => page.communitiesList),
+          hasMore: query.hasNextPage,
+        },
+      ]);
+      setSelected([...selected.slice(0, level), community]);
+    } else {
+      if (community.communityId === selected[level].communityId) {
+        //a previously selected item is clicked, so unselect it
+        //treat level = 0 as a special case
+        if (level === 0) {
+          setSelected([]);
+        } else {
+          setSelected([...selected.slice(0, level)]);
+        }
+        setCachedQueryResults(cachedQueryResults.slice(0, level));
+      } else {
+        //a previously unselected item is clicked
+        setSelected([...selected.slice(0, Math.max(level, 0)), community]);
+        setCachedQueryResults(cachedQueryResults.slice(0, level + 1));
+      }
+    }
+  };
+
+  return (
+    <div className={classes.root}>
+      {cachedQueryResults.map((query, index) => (
+        <BrowserColumn
+          key={index}
+          parent={selected?.[index - 1] ?? globalCommunityQuery.data}
+          communities={query.data}
+          handleClick={(community) => handleClick(community, index)}
+          selected={selected[index]?.communityId}
+        />
+      ))}
+      {query.isLoading ? ( // div prevents overflow scrollbar from spinner
+        <div className={classes.loader}>
+          <CircularProgress />
+        </div>
+      ) : query.isSuccess && globalCommunityQuery.isSuccess ? (
+        <div ref={lastColumnRef}>
+          <BrowserColumn
+            parent={
+              selected?.[selected.length - 1] ?? globalCommunityQuery.data
+            }
+            communities={query.data.pages.flatMap(
+              (page) => page.communitiesList
+            )}
+            handleClick={(community) => handleClick(community, selected.length)}
+          />
+          {query.hasNextPage && (
+            <Button
+              onClick={() => query.fetchNextPage()}
+              loading={query.isFetchingNextPage}
+              variant="outlined"
+            >
+              {LOAD_MORE}
+            </Button>
+          )}
+        </div>
+      ) : (
+        <Alert severity="error">{query?.error?.message || ""}</Alert>
+      )}
+    </div>
+  );
+}
+
+function BrowserColumn({
```

Comment:
For parent communities that have no child communities (like "Europe, Africa and Middle East" in next):
![Screenshot 2021-09-19 at 01 28 33](https://user-images.githubusercontent.com/13669362/133911800-dd0ecdb2-2894-4db9-aa2d-cc44a031fadb.png)

Might be worth adding an empty state message and a test for the message in that state? Otherwise, might look a bit confusing / as if it's broken even though it's working